client/src/components/Draftail/sources/ModalWorkflowSource.js

```diff
@@ -176,13 +176,13 @@ class ImageModalWorkflowSource extends ModalWorkflowSource {
 
     if (entity) {
       const data = entity.getData();
-      url = `${global.chooserUrls.imageChooser}${data.id}/select_format/`;
+      url = `${this.props.entityType.chooserUrls.imageChooser}${data.id}/select_format/`;
```

Comment:
Similar to other comments already but a reminder that we will need to support both approaches, first looking for the 'core' code approach of passing the data in and then also checking the global.

So each of these chooserUrls values will need to first check in the `this.props.chooserUrls` and then check for the same value in global.chooserUrls

Reply:
Thanks ,I have made the changes but I just surprised because you ask me to remove  editor_js hooks

> A reminder that you will want to remove the editor_js hooks used by embeds/documents/images addition of the
  globals in the

and also here you ask me to check about global I just want to know where global come from ?
I already removed the editor_js` hooks as you ask me to do. or global come from third-party packages ? ,
really I don ,t know, but I made the changes .